### ISSUE TITLE
SNMP proxy forwarder

### ISSUE THREAD

**Issue by markabrahams** [markabrahams/node-net-snmp#59]

**Comment by markabrahams** [comment]
I've implemented proxy forwarding for the agent, released in npm v2.4.0.

Use like this:
```
var forwarder = agent.getForwarder ();
forwarder.addProxy({
    context: "slatescontext",
    host: "bedrock",
    user: {
        name: "slate",
        level: snmp.SecurityLevel.authNoPriv,
        authProtocol: snmp.AuthProtocols.sha,
        authKey: "quarryandgravel"
    },
});
```

Then hit your proxy with a local agent user, e.g. add local user fred to the agent, and use fred to access the proxy:

```
var authorizer = agent.getAuthorizer();
authorizer.addUser ({
    name: "fred",
    level: snmp.SecurityLevel.noAuthNoPriv
});

// Test access using Net-SNMP tools (-n is the context option):

snmpget -v 3 -u fred -l noAuthNoPriv -n slatescontext localhost
```

Not rocket surgery.

**Comment by markabrahams** [comment]
That pretty much completes library support (at least an initial effort!) for all the RFC 3413 application types:

1. Command Generator
2. Command Responder
3. Notification Originator
4. Notification Receiver
5. Proxy Forwarder
